flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -999,9 +999,16 @@ class ScalarExprLowering {
     auto baseString = std::visit(
         Fortran::common::visitors{
             [&](const Fortran::evaluate::DataRef &x) { return gen(x); },
-            [&](const Fortran::evaluate::StaticDataObject::Pointer &)
+            [&](const Fortran::evaluate::StaticDataObject::Pointer &p)
                 -> fir::ExtendedValue {
-              TODO(getLoc(), "StaticDataObject::Pointer substring");
+              if (auto str = p->AsString())
+                return Fortran::lower::createStringLiteral(builder, getLoc(),
+                                                           *str);
+              // TODO: convert StaticDataObject to Constant<T> and use normal
+              // constant path. Beware that StaticDataObject data() takes into
+              // account target endianness.
```

Comment:
Do you actually mean target endianness or build machine endianness?

Reply:
I think you are right and this is the build machine endianness. I have updated the comment.